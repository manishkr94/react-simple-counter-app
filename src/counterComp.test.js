import React from 'react';
import renderer from 'react-test-renderer';
import { App } from './App';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
//import Adapter from 'enzyme-adapter-react-16';

enzyme.configure({ adapter: new Adapter() });

describe('Test counter', () => {
    test('render counter component', () => {
        const props = {
            count: 0,
            increaseCounter: jest.fn(),
            decreaseCounter: jest.fn()
        }
        const tree = renderer.create(
            <App {...props} />
        ).toJSON();
        expect(tree).toMatchSnapshot();

        const component2 = mount(
            <App {...props} />
        )
        expect(component2.find(".count").exists());
    });
});