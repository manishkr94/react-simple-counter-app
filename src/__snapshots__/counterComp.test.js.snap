// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test counter render counter component 1`] = `
<div
  className="App"
>
  ;
  <div
    className="count"
  >
    Count: 
    0
  </div>
  <button
    onClick={[Function]}
  >
    Increase Count
  </button>
  <button
    onClick={[Function]}
  >
    Decrease Count
  </button>
</div>
`;
